tests/contracts/solidity/ECContracts.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.3;
+
+contract RecoveryChecker {
+    constructor() {}
+
+    function checkRecovery(
+        bytes32 msgHash,
+        uint256 v,
+        bytes32 r,
+        bytes32 s
+    ) public view returns (address) {
+        assembly {
+            let pointer := mload(0x40)
+
+            mstore(pointer, msgHash)
+            mstore(add(pointer, 0x20), v)
+            mstore(add(pointer, 0x40), r)
+            mstore(add(pointer, 0x60), s)
+
+            if iszero(staticcall(not(0), 0x01, pointer, 0x80, pointer, 0x20)) {
```

Comment:
Why `not(0)`?
Note that you can use the syntax `address.staticcall(...)` which works like the `.call` version, and which don't require assembly.

Reply:
1) `not(0)` is residual copypasta from the [docs page](https://docs.moonbeam.network/builders/pallets-precompiles/precompiles/eth-mainnet/), which i'm not entirely sure why we have to include it
2) `v` is a uint8 usually, however to reproduce a bug we need to pass in a uint256 value in which required assembly (as built in `ecrecover()` gives type error). Would static calling the precompile addr been equivalent to this?

Reply:
`address(0x01).staticcall(abi.encode(msgHash, v, r, s))` should work?